File: CloudLocal/aws/localstack/pipeline/Node.js/Web/Plain/src/cache/Adapters/Redis.js
const Redis = require('ioredis');
const CacheAdapter = require('../CacheAdapter');

class RedisCache extends CacheAdapter {
  constructor() {
    super();
    this._client = null;
  }

  _getClient() {
    if (!this._client) {
      this._client = new Redis({
        host: process.env.REDIS_HOST || 'localhost',
        port: parseInt(process.env.REDIS_PORT || '6379'),
        maxRetriesPerRequest: null,
      });
    }
    return this._client;
  }

  async get(key) {
    const client = this._getClient();
    const value = await client.get(key);
    return value ? JSON.parse(value) : null;
  }

  async set(key, value, ttl = 300) {
    const client = this._getClient();
    await client.set(key, JSON.stringify(value), 'EX', ttl);
  }

  async del(key) {
    const client = this._getClient();
    await client.del(key);
  }

  async increment(key, ttl = 60) {
    const client = this._getClient();
    const value = await client.incr(key);
    if (value === 1) {
      await client.expire(key, ttl);
    }
    return value;
  }
}

module.exports = { RedisCache };

File: Inboxes/Python/Web/Plain/src/cache/adapters/redis.py
import os
import json
from cache.cache_adapter import Cache

class Redis(Cache):
    def __init__(self):
        import redis as r
        host, port = os.getenv("REDIS_HOST", "localhost:6379").split(":")
        self._redis = r.Redis(host=host, port=int(port), socket_connect_timeout=1)
        self._redis.ping()

    def get(self, key):
        val = self._redis.get(key)
        return json.loads(val) if val else None

    def set(self, key, value, ttl=300):
        self._redis.setex(key, ttl, json.dumps(value))

    def delete(self, key):
        self._redis.delete(key)
